test/focus-first-suggestion/AutosuggestApp.js

```diff
@@ -4,46 +4,52 @@ import Autosuggest from '../../src/Autosuggest';
 import languages from '../plain-list/languages';
 import { escapeRegexCharacters } from '../../demo/src/components/utils/utils.js';
 
-const getMatchingLanguages = value => {
+const getMatchingLanguages = (value) => {
   const escapedValue = escapeRegexCharacters(value.trim());
   const regex = new RegExp('^' + escapedValue, 'i');
 
-  return languages.filter(language => regex.test(language.name));
+  return languages.filter((language) => regex.test(language.name));
 };
 
 let app = null;
 
-export const getSuggestionValue = sinon.spy(suggestion => {
+export const getHighlightFirstSuggestion = sinon.spy(
+  (highlightFirstSuggestion, value) => {
+    return Boolean(highlightFirstSuggestion && value);
+  }
+);
```

Comment:
This doesn't make much sense I'm afraid, I think there's a misunderstanding somewhere.

This code defines a function, which you manually call in the tests, but it doesn't have any side effects, so that doesn't do anything. You then pass this function as the `highlightFirstSuggestion` prop below, which is interpreted as truthy, because it's a function, but it's actually never called there. Maybe `sinon.spy` is creating some confusion: all that does is record every call to this function, so you can check in tests whether the function was called.

If you replace this with just `getSuggestionValue = () => {}` it still does the same thing (and the tests still pass), and similarly if you keep this but swap the `true`/`false` values in the test that passes too.

I think you want to toggle the `highlightFirstSuggestion` prop, so that you actually pass `true` or `false` to `<Autosuggest>` below, and control it from the tests, right? You can do that by:

* Exporting a function here that runs `app.setState({ highlightFirstSuggestion: /* true / false */})`
* Using `this.state.highlightFirstSuggestion` as the `highlightFirstSuggestion` prop in `render()` below
* Updating the test code to call your exported function when you want to switch the values.

Does that make sense? Happy to answer any questions.